src/invidious/helpers/extractors.cr

```diff
@@ -0,0 +1,562 @@
+# This file contains helper methods to parse the Youtube API json data into
+# neat little packages we can use
+
+# Tuple of Parsers/Extractors so we can easily cycle through them.
+private ITEM_CONTAINER_EXTRACTOR = {
+  Extractors::YouTubeTabs,
+  Extractors::SearchResults,
+  Extractors::Continuation,
+}
+
+private ITEM_PARSERS = {
+  Parsers::VideoRendererParser,
+  Parsers::ChannelRendererParser,
+  Parsers::GridPlaylistRendererParser,
+  Parsers::PlaylistRendererParser,
+  Parsers::CategoryRendererParser,
+}
+
+record AuthorFallback, name : String? = nil, id : String? = nil
```

Comment:
wouldn't it be better to only accept `String`s, defaulting to `""`, so we don't have to do that sort of things afterwards? (taken from L53-54):
```crystal
author = author_fallback.name || ""
author_id = author_fallback.id || ""
```

Reply:
Having that distinction looks better imo, but I've went ahead and pushed 43ea8fa70698ef94701fdf9da419300b9a6a0710 which should resolve that.

Reply:
Imho, it makes it harder to read. And I don't know how crystal handles that, but less unions is probably better for performance (generated code is less complex).